pkg/k8s/controllers/telemetryservice_controller.go

```diff
@@ -68,3 +94,78 @@ func (r *TelemetryServiceReconciler) SetupWithManager(mgr ctrl.Manager) error {
 		For(&v1alpha1.TelemetryService{}).
 		Complete(r)
 }
+
+func CreateServiceIfNotExists(ctx context.Context, r *TelemetryServiceReconciler, ts *v1alpha1.TelemetryService, req ctrl.Request) error {
```

Comment:
It appears that this function only create telemetryservice service but does not check if it already exists. How about renaming it to `CreateTelemetryServiceService`?